### DISCUSSION TITLE
Extract new keys from code and tag with CLI

### DISCUSSION THREAD

**Discussion by SashaTrisor** [tolgee/tolgee-platform#2797] [Q&A] [answered]
I want to add and tag new keys in the code during a CI pipeline run. After reading the documentation, I thought I could tag them as "draft" using the CLI when introducing new keys in the code. However, after trial and test, I think I might be missing something:

- When I use `tagNewKeys: ['draft']` in the React SDK init and sync, the `draft` tag is only visible in the Chrome extension UI.
- When I run `tolgee sync`, new keys are created on the platform, but I don't have a way to apply a tag in the sync command.
- When I use `tolgee tag --filter-not-extracted draft`, I don't have the new keys on the platform.
- When I use `tolgee push` (which allows the `--tag-new-keys` option), it works with translation files, not the source code.

Am I missing a `tolgee sync --tag-new-keys` option, or is there a specific sequence of sync and tag commands I could use?

**Comment by stepan662** [comment]
Hello, unfortunately this function is currently not available. However it make sense to add it to the sync command, we already have an issue to do this https://github.com/tolgee/tolgee-cli/issues/126
I'll look into it as soon as possible.

**Reply by stepan662** [reply, marked as answer]
It was released in version 2.5.0
